src/driver-hidpp10.c

```diff
@@ -448,7 +448,7 @@ hidpp10drv_read_profile(struct ratbag_profile *profile)
 		free(profile->name);
 		profile->name = NULL;
 	}
-	if (p.name && p.name[0] != '\0')
+	if (*p.name && p.name[0] != '\0')
```

Comment:
Shouldn't this be `p.name != NULL`?

Reply:
Null check is always implied.

Reply:
That seems a weird to me, but if `p.name` is indeed never null, then we can just remove the `p.name` check. Replacing `p.name` with `*p.name` will just instead check the first character of the name.

Reply:
Hi, you are right. I removed the check.